packages/notifi-dapp-example/src/hooks/useNotifiRouter.ts

```diff
@@ -15,23 +15,30 @@ export const useNotifiRouter = () => {
     return frontendClientStatus.isInitialized;
   }, [frontendClientStatus]);
   const { handleRoute, isLoadingRouter } = useRouterAsync();
-  const { setIsGlobalLoading } = useGlobalStateContext();
+  const { setIsGlobalLoading, seIisInitialized, isInitialized } =
+    useGlobalStateContext();
 
   const { ftuStage, syncFtuStage } = useNotifiSubscriptionContext();
 
   const { cardConfig } = useNotifiCardContext();
   useEffect(() => {
+    seIisInitialized('initializing');
     if (frontendClientStatus.isAuthenticated) {
-      syncFtuStage(cardConfig.isContactInfoRequired);
+      syncFtuStage(cardConfig.isContactInfoRequired).finally(() => {
+        seIisInitialized('initialized');
+      });
     }
+    seIisInitialized('initialized');
```

Comment:
Using the logic in `useNotifiRouter` to change the GlobalState IMO is not good for `single responsibility priciple`